js/manager.js

```diff
@@ -1,149 +1,144 @@
+* Copyright (c) 2018, Voila contributors                                   *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+import { RenderMimeRegistry, standardRendererFactories } from '@jupyterlab/rendermime';
+import { WidgetManager as JupyterLabManager } from '@jupyter-widgets/jupyterlab-manager/lib/manager';
+import { HTMLManager, requireLoader } from '@jupyter-widgets/html-manager';
+import { WidgetRenderer } from '@jupyter-widgets/jupyterlab-manager/lib/renderer';
+import * as output from '@jupyter-widgets/jupyterlab-manager/lib/output';
+import * as base from '@jupyter-widgets/base';
+import * as controls from '@jupyter-widgets/controls';
+
+if (typeof window !== "undefined" && typeof window.define !== "undefined") {
+    window.define("@jupyter-widgets/base", base);
+    window.define("@jupyter-widgets/controls", controls);
+    window.define("@jupyter-widgets/output", output);
+  }
+
+const WIDGET_MIMETYPE = 'application/vnd.jupyter.widget-view+json';
+const htmlManager = new HTMLManager({ loader: requireLoader });
+
+export class WidgetManager extends JupyterLabManager {
+
+    constructor(kernel) {
+        const context = createContext(kernel);
+        const rendermime = createRenderMimeRegistry();
+        super(context, rendermime);
+        this._registerWidgets();
+        this.loader = requireLoader;
+    }
+
+    async build_widgets() {
+        const models = await this._build_models();
+        const tags = document.body.querySelectorAll('script[type="application/vnd.jupyter.widget-view+json"]');
+        for (let i=0; i!=tags.length; ++i) {
+            const viewtag = tags[i];
+            const widgetViewObject = JSON.parse(viewtag.innerHTML);
+            const { model_id } = widgetViewObject;
+            const model = models[model_id];
+            const widgetTag = document.createElement('div');
+            widgetTag.className = 'widget-subarea';
+            viewtag.parentElement.insertBefore(widgetTag, viewtag);
+            const view = await this.display_model(undefined, model, { el : widgetTag });
+            widgetTag.appendChild(view.node);
+        }
+    }
+
+    async loadClass(className, moduleName, moduleVersion) {
+        if (
+            moduleName === '@jupyter-widgets/base' ||
+            moduleName === '@jupyter-widgets/controls' ||
+            moduleName === '@jupyter-widgets/output'
+        ) {
+            return super.loadClass(className, moduleName, moduleVersion);
+        }
+        else {
+            return htmlManager.loadClass(className, moduleName, moduleVersion);
+        }
+    }
+
+    _registerWidgets() {

```

Comment:
This method exists because the jupyterlab-manager API does not expose the plugin extension directly. I don't know how keen they are to change their API. Given that I see recurrent names of people in all jupyter related repositories I hope this comment can be picked up.

Reply:
I'm not sure I follow you, this looks fine to me. I guess you see a reason not to have this code, could you explain?

Reply:
Sorry I realize my comment was not clear. This piece of code exists in the jupyterlab-manager package and a slight modification of their API could allow us to fully consume their module without replicating some of the code.